aten/src/ATen/core/TensorImpl.h

```diff
@@ -1459,6 +1519,13 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
   bool is_contiguous_ = true;
   bool is_variable_ = false;
   bool is_wrapped_number_ = false;
+
+  // We need this field because we want to prevent users from changing tensor metadata
+  // (e.g. sizes / strides / storage / storage_offset) of a derived tensor (i.e. tensors
+  // created from Python `tensor.data` or Python/C++ `tensor.detach()`), because those changes
+  // will not update the original tensor.
```

Comment:
This is like saying, "We want to prevent users from doing X, because they will not 'thing that obviously wouldn't happen if you did X'." The reason why they might get confused is because it *used* to update the old tensor, which is what we should say here.